DeveloperGuide.adoc: Update doc for Storage (#86)
CS2103-AY1819S1-T16-4/main: docs/DeveloperGuide.adoc

* `INFO` : Information showing the noteworthy actions by the App
* `FINE` : Details that is not usually noteworthy but may be useful in debugging e.g. print the actual list instead of just its size

== Module Information

Information of a NUS module such as module code, module credit and prerequisite modules are stored in `ModuleInfo` class.
This class is made to be immutable to avoid accidental changes to the class.

A python script (link:{repoURL}/tools/moduleConverter.py[`moduleConverter.py`]) is used to download the list of NUS modules
from NUSMods server, do some data transformation and write to `data/moduleInfo.json` for the use of our app.

> NUSMods provides the prerequisites and preclusion of a module in normal English sentence (e.g. "CS1010 or its equivalents"

Reviewer: ```suggestion
Since NUSMods provides the prerequisites and preclusions of a module in normal English sentence (e.g. "CS1010 or its equivalents"
```

and "Departmental approval"), attempting to extract the correct prerequisite and preclusion relationship of the modules
would be infeasible due to the potential long list of exception rules. Therefore, we will only extract the module codes

> in the sentences with regular expression, downside of this is that we won't include CS1010S, CS1010J and so on from

Reviewer: ```suggestion
in the sentences with regular expression, with the downside being that we won't include CS1010S, CS1010J and so on from
```

"CS1010 or its equivalents". We will also assume that if any module in the prerequisite/preclusion list is matched,
the condition is met immediately (even though this is incorrect for rules like "(MA1101R AND MA1221) OR MA1521").

> `JsonModuleInfoStorage` opens the processed `data/moduleInfo.json` file and deserialize its content into an array of

Reviewer: The jump to this sentence seems kinda sharp

> `ModuleInfo` objects. As JSON can only hold simple values such as strings and numbers, prerequisites and preclusion

Reviewer: ```suggestion
`ModuleInfo` objects. As JSON can only hold simple values such as strings and numbers, prerequisites and preclusions
```
